enterprise/frontend/src/embedding-sdk/sdk-wrapper/hooks/private/use-lazy-selector.ts

```diff
@@ -0,0 +1,22 @@
+import { useSyncExternalStore } from "react";
+
+import { useMetabaseProviderPropsStore } from "embedding-sdk/sdk-shared/hooks/use-metabase-provider-props-store";
+import type { SdkStoreState } from "embedding-sdk/store/types";
+
+const noop = () => {};
+
+export function useLazySelector<TSelected>(
+  selector: ((state: SdkStoreState) => TSelected) | null | undefined,
+): TSelected | null {
+  const {
+    props: { reduxStore },
+  } = useMetabaseProviderPropsStore();
+
+  const subscribe = (notify: () => void) =>
+    reduxStore && selector ? reduxStore.subscribe(notify) : noop;
+
+  const getSnapshot = () =>
+    reduxStore && selector ? selector(reduxStore.getState()) : null;
+
+  return useSyncExternalStore(subscribe, getSnapshot, () => null);
+}
```

Comment:
The selector hook that returns `null` until the SDK bundle is loaded, so all SDK public hooks can be called even before the SDK bundle is loaded